org.metaborg.spoofax.shell.console/src/main/java/org/metaborg/spoofax/shell/client/console/impl/ConsoleRepl.java

```diff
@@ -35,19 +37,42 @@
      */
     @Inject
     public ConsoleRepl(IEditor editor, IDisplay display, ICommandInvoker invoker) {
-        super(invoker);
+        this.invoker = invoker;
         this.editor = editor;
         this.display = display;
     }
 
-    @Override
+    /**
+     * Whether or not to keep running the loop.
+     *
+     * @param running
+     *            {@code true} to keep the loop running, {@code} false to stop it.
+     *
+     * @see #run()
+     */
+    public void setRunning(boolean running) {
+        this.running = running;
+    }
+
+    /**
+     * Run {@code eval(read())} in a loop, for as long as {@code running} is {@code true}.
+     *
+     * @throws MetaborgException
+     *             When something goes wrong during execution.
+     * @throws CommandNotFoundException
+     *             When the command could not be found.
+     *
+     * @see #read()
+     * @see #eval(String)
+     * @see #setRunning(boolean)
+     */
     public void run() {
         try {
             this.editor.history().loadFromDisk();
 
             String input;
             setRunning(true);
-            while (running && (input = read()) != null) {
+            while (running && (input = editor.getInput()) != null) {
```

Comment:
Because of the removal of `IOException` in the `IEditor` interface, the private method below is not needed anymore.